napari/_qt/qt_main_window.py

```diff
@@ -384,6 +429,137 @@ def show_notification(notification: Notification):
         """Show notification coming from a thread."""
         NapariQtNotification.show_notification(notification)
 
+    # --- Updates
+    # ------------------------------------------------------------------------
+    def _check_updates(self, update_info):
+        """Check if an update is needed based on settings and show a
+        confirmation dialog, start the update process, skip or display
+        an information message box.
+
+        Parameters
+        ----------
+        update_info: dict
+            Dictionary with information on updates incluyding version,
+            installer type and installed versions found.
+        """
+        print(update_info)
+        _settings = get_settings()
+        startup = self._update_worker._startup
+        self._update_info = update_info
+        self._update_version = update_info["latest"]
+        # FIXME: To test failure
+        self._update_version = '0.4.16'
+        # FIXME: To test success
+        # self._update_version = '0.4.15'
```

Comment:
To test locally you need to uncomment this @DragaDoncila 

`self._update_version = '0.4.15'`

If you are running in a conda environment that should do it

Reply:
@goanpeca thanks for the dev instructions! I can see the update preferences and access the troubleshooting dialog, but I can't seem to get `Check for updates` to do anything. It neither prompts for an update nor displays the "You have the latest version" dialog.